File: src/main/java/com/drunkensoftball/api/team/repo/TeamRepository.java
package com.drunkensoftball.api.team.repo;

import com.drunkensoftball.api.team.domain.Team;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.jpa.repository.JpaRepository;

import java.util.List;

public interface TeamRepository extends JpaRepository<Team, Long> {

    Team findByUuid(String uuid);

    Team findByUuidAndManagerId(String uuid, Long managerUserId);

    List<Team> findByManagerId(Long managerUserId, Pageable pageable);

}

File: src/main/kotlin/com/drunkensoftball/api/team/repo/TeamRepository.kt
package com.drunkensoftball.api.team.repo

import com.drunkensoftball.api.team.domain.Team
import org.springframework.data.domain.Pageable
import org.springframework.data.jpa.repository.JpaRepository

interface TeamRepository : JpaRepository<Team, Long> {

    fun findByUuid(uuid: String): Team

    fun findByUuidAndManagerId(uuid: String, managerUserId: Long?): Team

    fun findByManagerId(managerUserId: Long?, pageable: Pageable): List<Team>

}
